ideascube/serveradmin/catalog.py

```diff
@@ -816,6 +816,34 @@ def _persist_catalog(self):
         persist_to_file(self._catalog_cache, self._available)
         persist_to_file(self._installed_storage, self._installed)
 
+    def _update_installed_metadata(self):
+        # These are the keys we must only ever update with an actual package
+        # upgrade, and thus not here.
+        blacklist = (
+            'sha256sum',
+            'type',
+            'version',
+        )
```

Comment:
I think we should only have a `checkum` key in the future, with a value containing the used algorithm, something like: `checksum: "sha256:xxxxx"`